src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -126,6 +125,9 @@ public void onMessage(String message) {
                 public void onClose(int code, String reason, boolean remote) {
                     log.info("Connection closed! Code: " + code + ", Reason: " + reason);
                     state = State.Disconnected;
+                    if (!userRequestedDisconnect) {
+                        sleepThenReconnect();
+                    }
                     if (!remote) {
```

Comment:
How does the change affect the rest of this method? After `sleepThenReconnect()`, the `StreamrClient.this.onClose()` would still get called, if `!remote`? 

Is there an opportunity to make the previously existing code clearer here? What is `remote` and does it overlap in meaning with `userRequestedDisconnect`? Does `StreamrClient.this.onClose()` get called from anywhere else?

Reply:
Not intentional.

I had trouble finding the JavaDoc but eventually found it via google, according to it: https://www.javadoc.io/doc/org.java-websocket/Java-WebSocket/1.3.8/org/java_websocket/client/WebSocketClient.html
> remote - Returns whether or not the closing of the connection was initiated by the remote host.

I guess I could get rid of the field `boolean userRequestedDisconnect` and assume that `remote == true` is indicative of a user's intention of disconnecting. At least based on method `StreamrClient#onClose` this is the case.

Reply:
Got rid of the boolean field in newest commit